### ISSUE TITLE
Feature Request: Option to disable fallback to showing all line diagnostics when cursor is not over any diagnostic

### ISSUE THREAD

**Issue by RyanSaxe** [rachartier/tiny-inline-diagnostic.nvim#156]
## Problem Description

Currently, when the cursor is on a line with diagnostics but NOT directly over any diagnostic range, the plugin shows ALL diagnostics on that line. This happens even when `show_all_diags_on_cursorline = false`.

## Current Behavior

Looking at the source code in `lua/tiny-inline-diagnostic/filter.lua` (lines 36-40):

```lua
local current_pos_diags = vim.tbl_filter(function(diag)
  return diag.lnum == line and col >= diag.col and col <= diag.end_col
end, diagnostics)

return #current_pos_diags > 0 and current_pos_diags or diags_on_line
```

The issue is the fallback `or diags_on_line` - when no diagnostics are found under the cursor, it falls back to showing all diagnostics on the line.

## Expected/Desired Behavior

There should be a configuration option to control this fallback behavior. When enabled, the plugin should only show diagnostics when the cursor is directly over them, with no fallback to showing all line diagnostics.

## Proposed Solution

Add a new configuration option like `show_diags_only_under_cursor` or `disable_line_diagnostic_fallback` that would change the behavior to:

```lua
if opts.options.show_diags_only_under_cursor then
  return current_pos_diags  -- No fallback
else
  return #current_pos_diags > 0 and current_pos_diags or diags_on_line  -- Current behavior
end
```

## Use Case

This is particularly useful when working with lines that have multiple diagnostics. Users may want to see diagnostics only when they're actively examining a specific part of the code (cursor over the diagnostic), rather than being shown all diagnostics whenever the cursor is on the same line.

## Configuration Example

```lua
opts = {
  options = {
    show_all_diags_on_cursorline = false,
    show_diags_only_under_cursor = true,  -- New option
    -- other options...
  }
}
```

This would provide more granular control over when diagnostics are displayed, improving the user experience for those who prefer a less cluttered interface.

**Comment by rachartier** [comment]
Set

```lua
{
     options = {
          multilines = false,
     }
}
```
